libraries/AP_BattMonitor/AP_BattMonitor_Backend.h

```diff
@@ -49,6 +49,15 @@ class AP_BattMonitor_Backend
     // return true if cycle count can be provided and fills in cycles argument
     virtual bool get_cycle_count(uint16_t &cycles) const { return false; }
 
+    // return true if serial number can be provided and fills in serial number 
+    virtual bool get_serial_number(char *serial_number, uint8_t &serial_name_len) const { return false; };
```

Comment:
Trailing semi-colons here are a bit messy.  Elsewhere too.

Don't want to pass in a reference for the length.

```suggestion
    virtual bool get_serial_number(char *serial_number, uint8_t serial_name_len) const { return false; }
```

similarly elsewhere